src/core/ipc_handler.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+//! Handler for IPC connectivity
+use crate::error::{ClientErrorKind, Result};
+#[cfg(test)]
+use mockstream::SyncMockStream;
+use std::io::{Read, Write};
+use std::os::unix::net::UnixStream;
+use std::path::PathBuf;
+use std::time::Duration;
+
+#[cfg(not(test))]
+const DEFAULT_SOCKET_PATH: &str = "/tmp/security-daemon-socket";
+#[cfg(not(test))]
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(1);
+
+/// This trait is created to allow the iterator returned by incoming to iterate over a trait object
+/// that implements both Read and Write.
+pub trait ReadWrite: Read + Write {}
+// Automatically implements ReadWrite for all types that implement Read and Write.
+impl<T: Read + Write> ReadWrite for T {}
+
+/// Handler for any of the currently-supported IPC channels for
+/// communicating with Parsec.
+#[cfg_attr(not(test), derive(Debug))]
+#[derive(Clone)]
+// Allowing this since `derivative` fails to ignore the MockIpc
+#[allow(missing_debug_implementations)]
+pub enum IpcHandler {
```

Comment:
A new struct, `UnixSocket` could implement `Speak` in one file and another new struct `MockIpc` could implement it as well, in the test. That way we do not need the `#[cfg(test)]`